# Tugas membuat analisis Data menggunakan Jupyter Notebook (Python)

Nama : Muhammad Rifqi Riansyah Matondang

NIM  : 16520370

Universitas: Institut Teknologi Bandung

Tanggal mulai pengerjaan tugas: 15 November2020

Sumber dataset:https://www.kaggle.com/abcsds/pokemon


Diberikan dataset pokemon dan kekuatanya, kita akan mencari pokemon mana yang memiliki total stats paling tinggi dengan menjumlahkan poin ke 6 stats dan membandingkanya dengan pokemon lain. (Kecuali pokemon legendary)


## Menggunakan library dan memasukkan data ke notebook

In [28]:
import pandas as pd #mengimport library pandas #as pd artinya ketika kita mau mengakses library pandas kita dapat menggunakan pd instead of pandas

df = pd.read_csv('pokemon_data.csv')  #menggunakan df sebagai variable # disini kita menggunakan pd artinya kita mengaksesi library panda # pd.read_csv('pokemon_data.csv') artinya kita memasukan data berjenis csv ke notebook, dalam parentheses itu adalah nama file di folder saya.

df #mengakses variabel df

#                   Name   Type 1  Type 2  HP  Attack  Defense  \
0      1              Bulbasaur    Grass  Poison  45      49       49   
1      2                Ivysaur    Grass  Poison  60      62       63   
2      3               Venusaur    Grass  Poison  80      82       83   
3      3  VenusaurMega Venusaur    Grass  Poison  80     100      123   
4      4             Charmander     Fire     NaN  39      52       43   
..   ...                    ...      ...     ...  ..     ...      ...   
795  719                Diancie     Rock   Fairy  50     100      150   
796  719    DiancieMega Diancie     Rock   Fairy  50     160      110   
797  720    HoopaHoopa Confined  Psychic   Ghost  80     110       60   
798  720     HoopaHoopa Unbound  Psychic    Dark  80     160       60   
799  721              Volcanion     Fire   Water  80     110      120   

     Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0         65       65     45           1      False  
1         80       80     60           1      False  
2        100      100     80           1      False  
3        122      120     80           1      False  
4         60       50     65           1      False  
..       ...      ...    ...         ...        ...  
795      100      150     50           6       True  
796      160      110    110           6       True  
797      150      130     70           6       True  
798      170      130     80           6       True  
799      130       90     70           6       True  

[800 rows x 12 columns]

## Reading the data

In [9]:
#membaca kolom
print(df.columns) #membaca kolom dan headernya

Index(['#', 'Name', 'Type 1', 'Type 2', 'HP', 'Attack', 'Defense', 'Sp. Atk',
       'Sp. Def', 'Speed', 'Generation', 'Legendary'],
      dtype='object')


## Cleaning the data / Data preprocessing

Karena kita akan hanya akan membandingkan stats kekuatan maka kita hanya perlu kolom #, Nama, HP, Attack, Defense, Sp. Atk, Sp. Def, Speed. Maka dari itu kita dapat membuang Type 1, Type 2, Generation, Legendary


In [35]:
#df = df.drop(columns=['Type 1','Type 2','Generation', 'Legendary') #saya comment out karena udah saya run 
df
#terlihat data telah dibersihkan

#                   Name  HP  Attack  Defense  Sp. Atk  Sp. Def  Speed  \
0      1              Bulbasaur  45      49       49       65       65     45   
1      2                Ivysaur  60      62       63       80       80     60   
2      3               Venusaur  80      82       83      100      100     80   
3      3  VenusaurMega Venusaur  80     100      123      122      120     80   
4      4             Charmander  39      52       43       60       50     65   
..   ...                    ...  ..     ...      ...      ...      ...    ...   
795  719                Diancie  50     100      150      100      150     50   
796  719    DiancieMega Diancie  50     160      110      160      110    110   
797  720    HoopaHoopa Confined  80     110       60      150      130     70   
798  720     HoopaHoopa Unbound  80     160       60      170      130     80   
799  721              Volcanion  80     110      120      130       90     70   

     Total  
0      318  
1      405  
2      525  
3      625  
4      309  
..     ...  
795    600  
796    700  
797    600  
798    680  
799    600  

[800 rows x 9 columns]

## Sorting the data


Saya hanya menjumlahkan ke 6 stats dan membuat kolom baru berupa total, kemudian saya sort berdasarkan nilai total tertinggi dan dari situ kita dapat mengetahui pokemon mana yang paling hebat 

In [34]:
#df = df.drop(columns=['Type 1','Type 2','Generation', 'Legendary'])
df['Total']=df['HP']+df['Attack']+df['Defense']+df['Sp. Atk']+df['Sp. Def']+df['Speed']
df.sort_values('Total')


#                   Name   HP  Attack  Defense  Sp. Atk  Sp. Def  \
206  191                Sunkern   30      30       30       30       30   
322  298                Azurill   50      20       40       20       40   
446  401              Kricketot   37      25       41       25       41   
288  265                Wurmple   45      45       35       20       30   
16    13                 Weedle   40      35       30       20       20   
..   ...                    ...  ...     ...      ...      ...      ...   
424  383  GroudonPrimal Groudon  100     180      160      150       90   
422  382    KyogrePrimal Kyogre  100     150       90      180      160   
164  150    MewtwoMega Mewtwo Y  106     150       70      194      120   
426  384  RayquazaMega Rayquaza  105     180      100      180      100   
163  150    MewtwoMega Mewtwo X  106     190      100      154      100   

     Speed  Total  
206     30    180  
322     20    190  
446     25    194  
288     20    195  
16      50    195  
..     ...    ...  
424     90    770  
422     90    770  
164    140    780  
426    115    780  
163    130    780  

[800 rows x 9 columns]

## Data Analysis
Min, Max, Average

In [40]:
#df = df.drop(columns=['Type 1','Type 2','Generation', 'Legendary'])

df.mean(axis=0)


#          362.81375
HP          69.25875
Attack      79.00125
Defense     73.84250
Sp. Atk     72.82000
Sp. Def     71.90250
Speed       68.27750
Total      435.10250
dtype: float64

Bisa dilihat dari data sebelumnyabahwa 

max total power seri di 3 pokemon dan dimiliki oleh MewtwoMega Mewtwo X , RayquazaMega Rayquaza, MewtwoMega Mewtwo Y

kemudian min total power dimiliki oleh Sunkern

Average power dari 800 pokemon adalah 435.10250

## Data Exploration

saya akan mengambil beberapa type dan mencari average dari salah satu dari 6 stats pokemon dan mencoba mencari tren pokemon type mana yang memiliki rata2 paling tinggi di stats tertentu

In [51]:
newdf = pd.read_csv('pokemon_data.csv')
newdf.drop(columns=['Type 2', 'Generation', 'Legendary'])



#                   Name   Type 1  HP  Attack  Defense  Sp. Atk  \
0      1              Bulbasaur    Grass  45      49       49       65   
1      2                Ivysaur    Grass  60      62       63       80   
2      3               Venusaur    Grass  80      82       83      100   
3      3  VenusaurMega Venusaur    Grass  80     100      123      122   
4      4             Charmander     Fire  39      52       43       60   
..   ...                    ...      ...  ..     ...      ...      ...   
795  719                Diancie     Rock  50     100      150      100   
796  719    DiancieMega Diancie     Rock  50     160      110      160   
797  720    HoopaHoopa Confined  Psychic  80     110       60      150   
798  720     HoopaHoopa Unbound  Psychic  80     160       60      170   
799  721              Volcanion     Fire  80     110      120      130   

     Sp. Def  Speed  
0         65     45  
1         80     60  
2        100     80  
3        120     80  
4         50     65  
..       ...    ...  
795      150     50  
796      110    110  
797      130     70  
798      130     80  
799       90     70  

[800 rows x 9 columns]

In [53]:
TypeDrop= newdf[~(newdf['Type 1'] == 'Grass')].index 
Dropped = newdf.drop(TypeDrop)
newdf.drop(TypeDrop)
Dropped.mean(axis=0)

#             344.871429
HP             67.271429
Attack         73.214286
Defense        70.800000
Sp. Atk        77.500000
Sp. Def        70.428571
Speed          61.928571
Generation      3.357143
Legendary       0.042857
dtype: float64

In [55]:
TypeDrop= newdf[~(newdf['Type 1'] == 'Fire')].index 
Dropped = newdf.drop(TypeDrop)
newdf.drop(TypeDrop)
Dropped.mean(axis=0)

#             327.403846
HP             69.903846
Attack         84.769231
Defense        67.769231
Sp. Atk        88.980769
Sp. Def        72.211538
Speed          74.442308
Generation      3.211538
Legendary       0.096154
dtype: float64

In [56]:
TypeDrop= newdf[~(newdf['Type 1'] == 'Water')].index 
Dropped = newdf.drop(TypeDrop)
newdf.drop(TypeDrop)
Dropped.mean(axis=0)

#             303.089286
HP             72.062500
Attack         74.151786
Defense        72.946429
Sp. Atk        74.812500
Sp. Def        70.517857
Speed          65.964286
Generation      2.857143
Legendary       0.035714
dtype: float64

## Kesimpulan Data Exploration

Dari 3 type pokemon yang dianalasis, dapat disimpulkan bahwa

Rata-Rata HP tertinggi adalah Water

Rata-Rata Attack tertinggi adalah Fire

Rata-Rata Defense tertinggi adalah Water

Rata-Rata Sp. Atk tertinggi adalah Fire

Rata-Rata Sp. Def tertinggi adalah Fire

Rata-Rata Speed tertinggi adalah Fire